aziotctl/src/internal/check/checks/aziot_version.rs

```diff
@@ -98,11 +99,40 @@ impl AziotVersion {
                     }
                 }
             };
-            aziot_identity_service
+
+            let versions: Vec<String> = latest_versions
+                .channels
+                .iter()
+                .flat_map(|channel| channel.products.iter())
+                .filter(|product| product.id == "aziot-edge")
+                .flat_map(|product| product.components.iter())
+                .filter(|component| component.name == "aziot-identity-service")
+                .map(|component| component.version.clone())
+                .collect();
+
+            let actual_channel = Version::parse(actual_version)
+                .context("could not parse actual version as semver")?;
+            let expected_version = versions
+                .iter()
+                .find(|version| {
+                    let expected_channel = Version::parse(version)
+                        .context("could not parse expected version as semver")
+                        .unwrap(); // TODO: What's the right error handling here?
```

Comment:
I assume unwrap() is not a good option here, but not sure how best to handle this. Advice?

Reply:
`.try_find()` is unfortunately not stable, so you can do:

```rust
let versions: Vec<String> = ...;
let parsed_versions =
    versions.iter()
    .map(|version| Ok(Version::parse(version).context("...")?))
    .collect::<Result<Vec<_>>>()?;

...

let expected_version =
    parsed_versions.iter()
    .find(|expected_channel| expected_channel.major == ...);
```

Reply:
Fixed.